### PR TITLE
add support for SESv2 under feature flag

### LINKED ISSUE #229: Support of SESv2 client
SESv2 client (boto3/AWS) supports unsubscribe links and adds a new SQS event type.
This event tracks email provider options/buttons to unsubscribe from an email list.

### PR DESCRIPTION
**Pull request by fergusdixon**
Fixes #229

This is not adding support for any new features not previously supported by django-ses
This is just moving the current raw email logic to the new `send_email` method, which has some enhancements to cross-account verified identities under the hood

Personally, my motivation came from running into a similar issue to this https://github.com/aws/aws-cdk/issues/18964#issuecomment-1043009296

New features can be added in follow up PRs

I've tested this in my own environments and works end-to-end

### CODE REVIEW

File: django_ses/__init__.py [hunk 1]
```diff
@@ -250,6 +201,86 @@ def send_messages(self, email_messages):
...
+                sleep(delay)
+        recent_send_times.append(now)
+        # end of throttling
+
+    def _get_send_email_parameters(self, message, source):
+        return (self._get_v2_parameters(message, source)
+                if self._use_ses_v2
+                else self._get_v1_parameters(message, source))
+
+    def _get_v2_parameters(self, message, source):
```
Comment (fergusdixon) [inline django_ses/__init__.py:246]: [Relevant docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sesv2.html#SESV2.Client.send_email)

File: django_ses/__init__.py [hunk 2]
```diff
@@ -259,11 +290,22 @@ def get_rate_limit(self):
...
-            max_per_second = quota_dict['MaxSendRate']
-            ret = float(max_per_second)
-            cached_rate_limits[self._access_key_id] = ret
-            return ret
+            return self._get_v2_send_quota() if self._use_ses_v2 else self._get_v1_send_quota()
         finally:
             if new_conn_created:
                 self.close()
+
+    def _get_v2_send_quota(self):
```
Comment (fergusdixon) [inline django_ses/__init__.py:298]: [Relevant docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sesv2.html#SESV2.Client.get_account)

**Comment by fergusdixon** [comment]
> Thanks so much for your work on this, and adding tests! Only small comments.
> 
> Not sure what's going on with GitHub canceling the job runs, I'll try rerunning again later. Hopefully the tests will pass.

Thanks for the comments, will work on it today
As for the jobs:
Screenshot 2022-12-16 at 08 45 21

File: tests/test_backend.py [hunk 3]
```diff
@@ -148,6 +156,77 @@ def test_configuration_set_callable_send_mail(self):
         self.assertEqual(config_set_callable.dkim_selector, 'ses')
         self.assertEqual(config_set_callable.dkim_headers, ('From', 'To', 'Cc', 'Subject'))
 
+class SESV2BackendTest(TestCase):
+    def setUp(self):
+        # TODO: Fix this -- this is going to cause side effects
```
[the remark this answers is not in the document]
Comment (fergusdixon) [reply tests/test_backend.py:161]: This was kinda copied from the existing tests https://github.com/django-ses/django-ses/blob/24fee37f2600ea008b296fe958404df11f21d9ec/tests/test_backend.py#L94
https://github.com/fergusdixon/django-ses/blob/24fee37f2600ea008b296fe958404df11f21d9ec/tests/test_backend.py#L94
I feel I do a decent job cleaning up during the `tearDown`, so will remove the comment

**Comment by pcraciunoiu** [comment]
Merged in #269, releasing as v3.3.0 now

**Pull request closed without merging**